docs/security/README.md

```diff
@@ -0,0 +1,227 @@
+                    |       +----------------+     |4      +------+ 3a |      |
+                    |<------| OEK_encrypted' |<----+-------| EK_2 |<---|------+
+                    |       +----------------+             +------+    |      |
+                    |                    +----------------+         3b |      |
+                    |<-------------------| EK_2_encrypted |<-----------|------+
+                    |                    +----------------+            |
+                    |                                                  |
+
+
+1a) Send encrypted data key and master key ID to KMS.
+1b) Receive decrypted data key.
+2)  Decrypt encrypted object key with the KEK derived from the data key.
+3a) Receive new plain data key from the KMS using the master key ID of the server config.
+3b) Receive encrypted form of the data key from the KMS.
+4)  Derive a new KEK from the new data key and re-encrypt the OEK with it.
+5)  Store the encrypted OEK, encrypted data key and master key ID in object metadata.
+ ```
+<center>Figure 3 - KMS data key rotation</center>
+
+#### Key rotation - Extensions
+
+The basic SSE-S3 key rotation operation can be used to build more powerful key management operations.
+However, the following options are not implemented or completely designed, yet.
+
+1. **Master key migration**: The [SSE-S3 key rotation](#Key-rotation---Basic-Operation) can be performed
+                             on multiple/all objects to move them from one CMK to another CMK.
+2. **Secure object erasure**: The [SSE-S3 key rotation](#Key-rotation---Basic-Operation) can be applied
+                              to one/multiple objects with a randomly generated master key which is
+                              not stored at the KMS. That leads to an encrypted data key which can
+                              never be decrypted anymore.
+3. **Periodical key migration**: The [SSE-S3 key rotation](#Key-rotation---Basic-Operation) can be 
+                                 invoked after a certain time period to migrate one/some/all objects
+                                 from one master key to another.
+
+#### Secure Erasure and Locking
+
+The Minio server requires an available KMS to en/decrypt SSE-S3 encrypted objects. Therefore it
+is possible to erase or lock some or all encrypted objects. For example in case of a detected attack
+or other emergency situations the following actions can be taken:
+ - Seal the KMS such that it cannot be accessed by Minio server anymore. That will lock **all**
+   SSE-S3 encrypted objects protected by master keys stored on the KMS. All these objects can not
+   be decrypted as long as the KMS is sealed.
+ - Seal/Unmount one/some master keys. That will lock all SSE-S3 encrypted objects protected by
+   these master keys. All these object can not be decrypted as long as the key(s) are sealed.
+ - Delete one/some master keys. This is equal to erasing all SSE-S3 encrypted objects protected by
```

Comment:
"From a security standpoint, this is equivalent to..."  Minor point but need some separation to avoid implying that the objects are deleted from storage